version: '3.9'

services:
  app-tgbot:
    build:
      context: .
      dockerfile: tgbot.Dockerfile
    container_name: tgbot-container
    restart: unless-stopped
    env_file:
      - .env
    ports:
      - 80:80
#      - 443:443
      - 88:88
      - 8443:8443
    networks:
      - appnet
    volumes:
      - db-backups:${BACKUP_PATH}
    depends_on:
      app-db:
        condition: service_healthy

  app-db:
    build:
      context: .
      dockerfile: db.Dockerfile
      args:
        ADDITIONAL_LANGUAGE: ${ADDITIONAL_LANGUAGE}
    container_name: postgres-container
    restart: unless-stopped
    environment:
      TZ: ${TZ}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_INITDB_ARGS: ${POSTGRES_INITDB_ARGS}
    ports:
      - 5432:5432
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - appnet
    volumes:
      - db-data:/var/lib/postgresql/data
      - db-backups:/var/lib/postgresql/backups
      - .init-database.sql:/docker-entrypoint-initdb.d/init-database.sql

networks:
  appnet:
    name: app-net
    driver: bridge

volumes:
  db-data:
    name: db-data-volume
  db-backups:
    name: db-backups-volume
